pylsp_ruff/plugin.py

```diff
@@ -52,26 +52,44 @@
 def pylsp_settings():
     log.debug("Initializing pylsp_ruff")
     # this plugin disables flake8, mccabe, and pycodestyle by default
-    return {
+    settings = {
         "plugins": {
-            "ruff": {
-                "enabled": True,
-                "config": None,
-                "exclude": None,
-                "executable": "ruff",
-                "ignore": None,
-                "extendIgnore": None,
-                "lineLength": None,
-                "perFileIgnores": None,
-                "select": None,
-                "extendSelect": None,
-            },
+            "ruff": PluginSettings(),
             "pyflakes": {"enabled": False},
             "flake8": {"enabled": False},
             "mccabe": {"enabled": False},
             "pycodestyle": {"enabled": False},
         }
     }
+    return converter.unstructure(settings)
+
+
+@hookimpl(hookwrapper=True)
+def pylsp_format_document(workspace: Workspace, document: Document) -> Generator:
```

Comment:
mypy will complain without it.
reason being: the `@hookimpl(hookwrapper=True)` decorator turns the function into a generator

Reply:
not using `hookwrapper=True` here would lead to this formatting-hook not being executed if `python-lsp-black` is enabled tho:
- `python-lsp-server` defines `pylsp_format_document` to only execute the first hook (see [here](https://github.com/python-lsp/python-lsp-server/blob/develop/pylsp/hookspecs.py#L82))
- `python-lsp-black` registers its hook to be tried first (see [here](https://github.com/python-lsp/python-lsp-black/blob/master/pylsp_black/plugin.py#L31))

That's the reason why `pyls-isort` also uses `hookwrapper=True` (see [here](https://github.com/paradoxxxzero/pyls-isort/blob/master/pyls_isort/plugin.py#L48))

Reply:
Ah I see, that makes sense!

Reply:
@jhossbach are we waiting for a response from @ccordoba12 for a decision regarding `I` or do we just want to make a decision by tomorrow if @ccordoba12 doesnt reply?

Reply:
We can merge this PR first and then discuss whether to disable isort by default later. Let me open an issue for discussion

Reply:
@jhossbach so coming back to this after a week.
since there was no feedback until now i think we should just deactivate `pyls-isort` and default `format` to `["I"]`.